src/shadowsocks_config.ts

```diff
@@ -338,3 +338,63 @@ export const SIP002_URI = {
     return `ss://${userInfo}@${uriHost}:${port.data}/${queryString}${hash}`;
   },
 };
+
+export interface OnlineConfig {
+  // URL endpoint to retrieve a Shadowsocks configuration.
+  readonly url: string;
+  // Any additional configuration (e.g. `certFp`, `httpMethod`, etc.) may be stored here.
+  readonly extra: {[key: string]: string};
+}
+
+// Ref: https://github.com/shadowsocks/shadowsocks-org/issues/89
+export const SIP008_URI = {
+  PROTOCOL: 'ssconf',
+
+  validateProtocol: (uri: string) => {
+    if (!uri || !uri.startsWith(SIP008_URI.PROTOCOL)) {
+      throw new InvalidUri(`URI must start with "${SIP008_URI.PROTOCOL}"`);
+    }
+  },
+
+  parse: (uri: string): OnlineConfig => {
+    SIP008_URI.validateProtocol(uri);
+
+    // URL parser for expedience, replacing the protocol "ssconf" with "https" to ensure correct
+    // results, otherwise browsers like Safari fail to parse it.
+    const inputForUrlParser =
```

Comment:
Maybe simpler is 

    uri.replace(new RegExp(`^${SIP008_URI.PROTOCOL}`), https)

Reply:
done in 9a81716.